src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1359,6 +1359,13 @@ public void Dispose()
                 PreferenceSettings.MessageLogged -= LogMessage;
             }
 
+            if (!IsTestMode)
+            {
+                //The writer have to be disposed at DynamoModel level due that we could index package-nodes as new packages are installed
+                LuceneSearchUtility.writer?.Dispose();
+                LuceneSearchUtility.writer = null;
+            }
```

Comment:
I found that the `if (!IsTestMode)` has been repeating much many times in all the other places, since `LuceneSearchUtility` has a reference of `DynamoModel`, I would like to wrap the dispose or write code to be function on `LuceneSearchUtility` object, so that in other places we could just call `LuceneSearchUtility.DisposeWriter()` and not having to check `IsTestMode` all the time

Reply:
@QilongTang I've created two method in LuceneSearchUtility one for dispose the writer and the other for commiting changes, please let me know if this sounds good.
commit: https://github.com/DynamoDS/Dynamo/pull/14120/commits/85334fcdf0a658d74b5692fdad74005b7c6978a2